mne/time_frequency/tfr.py

```diff
@@ -1137,156 +1434,151 @@ def save(self, fname, overwrite=False):
-        out = AverageTFR(**tfr_dict[condition])
-    else:
-        out = [AverageTFR(**d) for d in list(zip(*tfr_data))[1]]
-    return out
+    # use union of bad channels
+    bads = list(set(tfr.info['bads']).union(*(t_.info['bads']
+                                              for t_ in all_tfr[1:])))
+    tfr.info['bads'] = bads
 
+    tfr.data = sum(w * t_.data for w, t_ in zip(weights, all_tfr))
+    tfr.nave = max(int(1. / sum(w ** 2 / e.nave
+                                for w, e in zip(weights, all_tfr))), 1)
+    return tfr
 
-@verbose
-def tfr_morlet(inst, freqs, n_cycles, use_fft=False, return_itc=True, decim=1,
-               n_jobs=1, picks=None, verbose=None):
-    """Compute Time-Frequency Representation (TFR) using Morlet wavelets
 
-    Parameters
-    ----------
-    inst : Epochs | Evoked
-        The epochs or evoked object.
-    freqs : ndarray, shape (n_freqs,)
-        The frequencies in Hz.
-    n_cycles : float | ndarray, shape (n_freqs,)
-        The number of cycles globally or for each frequency.
-    use_fft : bool
-        The fft based convolution or not.
-    return_itc : bool
-        Return intertrial coherence (ITC) as well as averaged power.
-        Must be ``False`` for evoked data.
-    decim : int | slice
-        To reduce memory usage, decimation factor after time-frequency
-        decomposition.
-        If `int`, returns tfr[..., ::decim].
-        If `slice` returns tfr[..., decim].
-        Note that decimation may create aliasing artifacts.
-        Defaults to 1.
-    n_jobs : int
-        The number of jobs to run in parallel.
-    picks : array-like of int | None
-        The indices of the channels to plot. If None all available
-        channels are displayed.
-    verbose : bool, str, int, or None
-        If not None, override default verbose level (see mne.verbose).
+# Utils
 
-    Returns
-    -------
-    power : instance of AverageTFR
-        The averaged power.
-    itc : instance of AverageTFR
-        The intertrial coherence (ITC). Only returned if return_itc
-        is True.
 
-    See Also
-    --------
-    tfr_multitaper, tfr_stockwell
-    """
-    decim = _check_decim(decim)
-    data = _get_data(inst, return_itc)
-    info = inst.info
-
-    info, data, picks = _prepare_picks(info, data, picks)
-    data = data[:, picks, :]
-
-    power, itc = _induced_power_cwt(data, sfreq=info['sfreq'],
-                                    frequencies=freqs,
-                                    n_cycles=n_cycles, n_jobs=n_jobs,
-                                    use_fft=use_fft, decim=decim,
-                                    zero_mean=True)
-    times = inst.times[decim].copy()
-    nave = len(data)
-    out = AverageTFR(info, power, times, freqs, nave, method='morlet-power')
-    if return_itc:
-        out = (out, AverageTFR(info, itc, times, freqs, nave,
-                               method='morlet-itc'))
-    return out
+def _get_data(inst, return_itc):
```

Comment:
WDYT about a (future) PR to extend this to include other classes within MNE? Could be a useful way to pull the data for an arbitrary MNE object and could be used in methods like `.crop` etc.